acorn/index.d.ts

```diff
@@ -26,43 +102,153 @@ declare namespace acorn {
+        _true: TokenType;
+        _false: TokenType;
+        _in: TokenType;
+        _instanceof: TokenType;
+        _typeof: TokenType;
+        _void: TokenType;
+        _delete: TokenType;
     }
 
+    class TokContext {
+        constructor(token: string, isExpr: boolean, preserveSpace: boolean, override: (p: Parser) => void);
+    }
+
+    const tokContexts: {
+        b_stat: TokContext;
+        b_expr: TokContext;
+        b_tmpl: TokContext;
+        p_stat: TokContext;
+        p_expr: TokContext;
+        q_tmpl: TokContext;
+        f_expr: TokContext;
+    };
+
+    function isIdentifierStart(code: number, astral?: boolean): boolean;
+
+    function isIdentifierChar(code: number, astral?: boolean): boolean;
+
     interface AbstractToken {
         start: number;
         end: number;
-        loc: ESTree.SourceLocation;
-        range: [number, number];
+        loc?: SourceLocation;
+        range?: [number, number];
+    }
+
+    interface Comment extends AbstractToken {
+        type: string;
+        value: string;
+    }
+
+    class Token {
+        constructor(p: Parser);
     }
 
     interface Token extends AbstractToken {
         type: TokenType;
         value: any;
     }
 
-    interface Comment extends AbstractToken {
-        type: string;
-        value: string;
+    function isNewLine(code: number): boolean;
+
+    const lineBreak: RegExp;
+
+    const lineBreakG: RegExp;
+
+    const version: string;
+
+    interface IParse {
+        (input: string, options?: Options): ESTree.Program;
     }
 
-    interface Options {
-        ecmaVersion?: number;
-        sourceType?: string;
-        onInsertedSemicolon?: (lastTokEnd: number, lastTokEndLoc?: ESTree.Position) => any;
-        onTrailingComma?: (lastTokEnd: number, lastTokEndLoc?: ESTree.Position) => any;
-        allowReserved?: boolean;
-        allowReturnOutsideFunction?: boolean;
-        allowImportExportEverywhere?: boolean;
-        allowHashBang?: boolean;
-        locations?: boolean;
-        onToken?: ((token: Token) => any) | Token[];
-        onComment?: ((isBlock: boolean, text: string, start: number, end: number, startLoc?: ESTree.Position, endLoc?: ESTree.Position) => any) | Comment[];
-        ranges?: boolean;
-        program?: ESTree.Program;
-        sourceFile?: string;
-        directSourceFile?: string;
-        preserveParens?: boolean;
-        plugins?: { [name: string]: Function; };
+    const parse: IParse;
+
+    function parseExpressionAt(input: string, pos: number, options?: Options): ESTree.Expression;
+
+    // todo: here the tokenizer function returns a Parser instance, that is targeting the detail of
+    // Parser prototype. Someone need this please reade README.md first.
+    // function tokenizer(options: Options, input: string): Parser;
+
+    let parse_dammit: IParse | undefined;
+    let LooseParser: LooseParser | undefined;
+    let pluginsLoose: PluginsObject | undefined;
+
+    interface LooseParser {
```

Comment:
Please mark this as `class`. Otherwise `addLooseExports`, which references `LooseParser` as an argument, will accept any empty object (`{}`) because in TypeScript interface are not tags, and any interface is compatible with any other with same properties, so empty interface is compatible with any empty object.

Also, `LooseParser` is indeed a `class` in Acorn source, so this will help to extend it in future more easily.